website/static/js/project-organizer.js

```diff
@@ -400,37 +334,75 @@ var tbOptions = {
-            // TODO: Dummy data, remove this when api is ready
-            item.contributors = [{
-                id: '8q36f',
-                name : 'Dummy User'
-            }];
-        });
-        console.log('Lazyload processed', value.data);
-        return value.data;
-    }
+        var node = item.data;
+        var linkObject = new LinkObject('node', node, node.attributes.title);
+        // Get ancestors
+        linkObject.ancestors = [];
+        function getAncestors (item) {
+            var parent = item.parent();
+            if(parent && parent.id > tb.treeData.id) {
+                linkObject.ancestors.unshift(parent);
+                getAncestors(parent);
+            }
+        }
+        getAncestors(item);
+        tb.options.updateFilesData(linkObject);
+    },
+    hScroll : 300,
+    filterTemplate : function() {
+        var tb = this;
+        return [
+            m('input.form-control[placeholder="Search all my projects"][type="text"]', {
+                style: 'display:inline;',
+                onkeyup: function(event){
+                    if ($(this).val().length === 1){
+                        tb.updateFolder(allProjectsCache().data, tb.treeData);
+                    }
+                    tb.filter(event);
+                },
+                onchange: m.withAttr('value', tb.filterText),
+                value: tb.filterText()
+            }),
+            m('.filterReset', { onclick : function () {
+                tb.resetFilter.call(tb);
+                $('#poFilter>input').val('');
+            } }, tb.options.removeIcon())
+        ];
+    },
+    hiddenFilterRows : ['tags']
 };
 
 var ProjectOrganizer = {
     controller : function (args) {
-        var poOptions = $.extend(
-            {
-                updateSelected : args.updateSelected,
-                updateFilesData : args.updateFilesData,
-                filesData: args.filesData()
-            },
-            tbOptions
-        );
         LinkObject = args.LinkObject;
-        this.tb = new Treebeard(poOptions, true);
+        var self = this;
+        self.updateTB = function(){
+            var poOptions = $.extend(
+                {
+                    updateSelected : args.updateSelected,
+                    updateFilesData : args.updateFilesData,
+                    filesData: args.filesData().data,
+                    wrapperSelector : args.wrapperSelector,
+                    dragContainment : args.dragContainment
+                },
+                tbOptions
+            );
+            var tb = new Treebeard(poOptions, true);
+            m.redraw.strategy('all');
+            return tb;
+        };
+        allProjectsCache = args.allProjects;
+        self.tb = self.updateTB();
     },
     view : function (ctrl, args) {
-        return m('.fb-project-organizer#projectOrganizer', ctrl.tb);
+        var tb = ctrl.tb;
+        if (args.reload()) {
+            tb = ctrl.updateTB();
+            args.reload(false);
```

Comment:
This is probably fine for now, but this can be improved by moving this update to the higher-level component (`FileBrowser`) which will make `ProjectOrganizer` mostly stateless.

Something along the lines of 

```js
var FileBrowser = {
    controller() {
        self.reload(false);
        self.handlePOUpdate = function() { self.reload(false); }
        self.shouldPOUpdate = function() {self.reload() === true; };
    }

    view(ctrl, ...) {
        
        m.component(ProjectOrganizer, {
            shouldUpdate: ctrl.shouldPOUpdate,
            onUpdate: ctrl.handlePOUpdate
        })
    }
}
```

```js
var ProjectOrganizer = {
    controller(args) {
        self.updateTB = function() {
            //...
            args.onUpdate.call(self);
        }
    },

    view(ctrl, args) {
        if (args.shouldUpdate()) {
            ctrl.updateTb()
        }
    }
}
```